File: Reto # - 10/Java/Excepciones.java
/**
 * @author DeathbatO
 */

public class Excepciones {
    public static void main(String[] args) {
        try {
            // Intenta dividir 10 por 0
            int resultado = 10 / 0;
        } catch (ArithmeticException e) {
            // Captura el error y muestra un mensaje personalizado
            System.out.println("Error: " + e.getMessage());
            System.out.println("No se puede dividir por cero.");
        }

        try {
            // Intenta acceder a un índice no existente de un listado
            int[] lista = {1, 2, 3};
            int elemento = lista[4];
        } catch (ArrayIndexOutOfBoundsException e) {
            // Captura el error y muestra un mensaje personalizado
            System.out.println("Error: " + e.getMessage());
            System.out.println("Índice fuera de rango.");
        }

        System.out.println("El programa continúa después de manejar las excepciones.");
    }
}

File: Reto # - 10/Python/Excepciones.py
## DeathbatO
# Manejo de excepciones en Python

try:
    # Intenta dividir 10 por 0
    resultado = 10 / 0
except ZeroDivisionError as error:
    # Captura el error y muestra un mensaje personalizado
    print("Error:", error)
    print("No se puede dividir por cero.")

try:
    # Intenta acceder a un índice no existente de un listado
    lista = [1, 2, 3]
    elemento = lista[4]
except IndexError as error:
    # Captura el error y muestra un mensaje personalizado
    print("Error:", error)
    print("Índice fuera de rango.")

print("El programa continúa después de manejar las excepciones.")
